### ISSUE TITLE
Fix false positives for negation within interpolation in declaration-bang-space-before

### ISSUE THREAD

**Issue by Benwick91** [stylelint/stylelint#4713]
> Clearly describe the bug

I had the following error after a stylelint update: 
`TypeError: Cannot read property 'stringify' of null`

In my research I find out, that I don't need the deprecated `stylelint-processor-styled-components`. So I delete the package and the error wasn't show anymore. But now stylelint also mark the logical "not" operator "!". Is there a possible to disable this behaviour? With the processor and version 13.0.0 it worked.

> Which rule, if any, is the bug related to?

`declaration-bang-space-before`

> What code is needed to reproduce the bug?

```
const Styles = styled.div<Props>`
  display: inline-block;
  border: 1px solid;
  border-color: ${(props: Props) =>
    !props.disabled ? '#000' : '#FFF'};
`
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "extends": [
    "stylelint-config-airbnb",
    "stylelint-config-styled-components",
    "stylelint-config-prettier"
  ],
  "plugins": ["stylelint-order"],
  "rules": {
    "order/order": [
      "custom-properties",
      "declarations"
    ],
  }
}
```

> Which version of stylelint are you using?

`13.3.3`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

"Yes, it's related to styled-components"

> What did you expect to happen?

No warnings/errors with "!" operator in conditional statements.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
Expected single space before "!"
```

Thank you!

**Comment by jeddy3** [comment]
@Benwick91 Thanks for the report and for using the template.

I can [reproduce this using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBc4IDsDOAXABAZUwXkwwE8AbGKAOgEMAeABQCcIAHVAPgAMAdZTTZpsxgN0ReJgAkwABQCWqcY3kBKfGx58+AQjmsKUAJaoqAIzJRMAfkwByAMQAGJzczj7AMU82AvgG4eHCAANCCoRMjoVAAeCOCoqAC0BsgJAFaowUjIAGYGAOaxwBqYXCAMAK5kqKXiRbyapbBgJFQMVOgGKAkmVMh5CajMVGAwCVTZ6MI1JSDIMABuU8HFfI0wza3tnSk9fQNDI90w2RAMMNOlVCQA7lRE1SDF3jzeIN5AA) with:

```js
const S = styled.a<Props>`
  property: ${(props: Props) =>
    !props.disabled ? '#000' : '#FFF'};
`
```

It's likely a problem with [the parser](https://github.com/stylelint/postcss-css-in-js) stylelint uses [to support CSS-in-JS](https://stylelint.io/about/syntaxes). I've added this issue to https://github.com/stylelint/stylelint/issues/4574, which is a growing list of the problems with the syntax that we need help to fix.

Please consider [contributing to the parser](https://stylelint.io/developer-guide/syntaxes) if you have time.

> Is there a possible to disable this behaviour?

Until the bug is resolved, you can [turn off the rule](https://stylelint.io/user-guide/configure#rules) using `null`.

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rule [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.
